metricbeat/module/system/raid/raid.go

```diff
@@ -82,12 +87,28 @@ func (m *MetricSet) Fetch(r mb.ReporterV2) {
 	}
 
 	for _, stat := range stats {
+		//stat.Name is the raid device name, which should be visible as /dev/md*
+		devicePath := path.Join(m.devFS, stat.Name)
+		dev, err := NewDevice(devicePath)
+		if err != nil {
+			r.Error(errors.Wrap(err, "failed to to open raid device for ioctl"))
+			return
+		}
+		defer dev.Close()
+		arrayInfo, err := dev.GetArrayInfo()
+		if err != nil {
+			r.Error(errors.Wrap(err, "failed to get data from md device"))
+			return
+		}
 		event := common.MapStr{
 			"name":           stat.Name,
 			"activity_state": stat.ActivityState,
 			"disks": common.MapStr{
-				"active": stat.DisksActive,
-				"total":  stat.DisksTotal,
+				"active":  arrayInfo.ActiveDisks,
+				"working": arrayInfo.WorkingDisks,
+				"failed":  arrayInfo.FailedDisks,
+				"spare":   arrayInfo.SpareDisks,
+				"total":   stat.DisksTotal,
```

Comment:
```suggestion
				"total":   arrayInfo.NrDisks,
```
?

Reply:
Ah, yah, that would work!

Reply:
I think the reason I left this out is because I wasn't 100% sure if I should use `RAIDDisks` or `NrDisks`I suspect it's the latter.